crates/test/macros/src/lib.rs

```diff
@@ -131,7 +131,7 @@ fn tests_runner_impl(args: TokenStream) -> parse::Result<TokenStream> {
             }
 
             // Test runner entry point
-            fn main(args: *mut u8) -> i32 {
+            pub(super) fn main(args: *mut u8) -> i32 {
```

Comment:
Since we are now explicitly calling this function, let's mark it `unsafe` describing its `# Safety` invariant (`args` validity requirements)

Reply:
I'd prefer to insulate this unsafety from users entirely by doing the argument-to-string parsing ourselves, as this function in user code is type-constrained by the `entry!` macro. I'll do that in a separate PR.

Reply:
#125